### PR TITLE
Enable openstack pods monitoring

### LINKED ISSUE #1995: [1] Create alerts for cloud-controller-manager
### Description

When working on an environment I noticed that the cloud-controller-manager had 0 replicas, because the control plane nodes on that environment had both `master` and `control-plane` taints and the cloud-controller-manager only had tolerations for the `control-plane` taint.

There should be some prometheus alerts that trigger if the cloud-controller-manager is not running.

### Additional context

This should also be done for the csi drivers.

### Definition of done

We have alerts letting us know when the cloud-controller-manager (and csi driver?) is not running.

### PR DESCRIPTION
**Pull request by crssnd**
> [!warning]
> **This is public repository, ensure not to disclose:**
> - [x] personal data beyond what is necessary for interacting with this pull request
> - [x] business confidential information, such as customer names

### What kind of PR is this?

**Required**: Mark one of the following that is applicable:

- [ ] kind/feature       
- [x] kind/improvement   
- [ ] kind/deprecation   
- [ ] kind/documentation 
- [ ] kind/clean-up      
- [ ] kind/bug           
- [ ] kind/other         

*Optional*: Mark one or more of the following that are applicable:

> [!important]
> Breaking changes should be marked `kind/admin-change` or `kind/dev-change` depending on type
> Critical security fixes should be marked with `kind/security`

- [x] kind/admin-change  
- [ ] kind/dev-change    
- [ ] kind/security      
- [ ] [kind/adr]()       

### Platform Administrator notice
The `openstack-monitoring` chart will be installed by default if you are running Kubernetes on an Openstack environment (`.openstackMonitoring.enabled = true` in `defaults/common-config.yaml`).
You need to make sure that `openstack-cloud-controller-manager` can expose metrics by setting adding the `--bind-address=0.0.0.0` flag.

### What does this PR do / why do we need this PR?

To collect metrics from `openstack-cloud-controller-manager` and also configure some alerting for this component.

- Fixes #1995 

#### Additional information to reviewers

1. Do we want the `openstack-cloud-controller-manager` version to follow Kubernetes version?
2. Let me know if you have a better solution to receive an alert when only the controller from only one cluster has disappeared. My current proposal is below, but has some limitations:
`(group(present_over_time(up{job="openstack-monitoring"}[6h])) by (cluster) unless group(up{job="openstack-monitoring"}) by (cluster)) or (absent(up{job="openstack-monitoring"} == 1))`

#### Checklist

- [x] Proper commit message prefix on all commits
  
- Change checks:
  - [x] The change is transparent
  - [ ] The change is disruptive
  - [x] The change requires no migration steps
  - [ ] The change requires migration steps
- Metrics checks:
  - [x] The metrics are still exposed and present in Grafana after the change
  - [ ] The metrics names didn't change (Grafana dashboards and Prometheus alerts are not affected)
  - [ ] The metrics names did change (Grafana dashboards and Prometheus alerts were fixed)
- Logs checks:
  - [x] The logs do not show any errors after the change
- Network Policy checks:
  - [x] Any changed pod is covered by Network Policies
  - [ ] The change does not cause any dropped packages in the `NetworkPolicy Dashboard`
- Pod Security Policy checks:
  - [ ] Any changed pod is covered by Pod Security Admission
  - [ ] Any changed pod is covered by Gatekeeper Pod Security Policies
  - [x] The change does not cause any pods to be blocked by Pod Security Admission or Policies
- Falco checks:
  - [x] The change does not cause any alerts to be generated by Falco
- Audit checks:
  - [ ] The change does not cause any unnecessary Kubernetes audit events
  - [ ] The change requires changes to Kubernetes audit policy
- Bug checks:
  - [ ] The bug fix is covered by regression tests

### CODE REVIEW

**Review by Elias-elastisys — APPROVED** [review]

File: helmfile.d/values/prometheus-alerts-sc.yaml.gotmpl [hunk 1]
```diff
@@ -35,6 +35,7 @@ defaultRules:
     networkpolicies: {{ .Values.networkPolicies.enableAlerting }}
     clusterApi: {{ .Values.clusterApi.enabled }}
     webhooks: {{ .Values.prometheus.webhookAlerts.enabled }}
+    openstack: {{ or (.Values | get "openstackMonitoring.enabled" false) (.Values | get "clusterApi.enabled") false }}
```
Comment (davidumea) [inline helmfile.d/values/prometheus-alerts-sc.yaml.gotmpl:38]: Shouldn't it be like this?
```suggestion
    openstack: {{ or (.Values | get "openstackMonitoring.enabled" false) (.Values | get "clusterApi.enabled" false) }}
```

File: helmfile.d/stacks/monitoring.yaml [hunk 2]
```diff
@@ -63,19 +63,21 @@ templates:
 
   openstack-monitoring:
     disableValidationOnInstall: true
-    condition: ck8sManagementCluster.enabled
-    installed: {{ .Values | get "openstackMonitoring.enabled" false }}
+    installed: {{ or (.Values | get "openstackMonitoring.enabled" false) (.Values | get "clusterApi.enabled" false) }}
```
Comment (davidumea) [inline helmfile.d/stacks/monitoring.yaml:66]: Did you consider if apps is being run on clusterAPI on a non-openstack provider?

Comment (crssnd) [reply helmfile.d/stacks/monitoring.yaml:66]: No, I had a discussion with Viktor and I should only keep `"openstackMonitoring.enabled" false` here
This value should apply for all openstack env, cluster-api and kubespray

File: helmfile.d/charts/openstack-monitoring/values.yaml [hunk 3]
```diff
@@ -3,3 +3,7 @@
 # Declare variables to be passed into your templates.
 openstackMonitoring:
   enabled: false
+serviceMonitor:
+  labels:
+    k8s-app: openstack-cloud-controller-manager
+  namespace: ~
```
Comment (Xartos) [inline helmfile.d/charts/openstack-monitoring/values.yaml:9]: question: Why is this `~`, wouldn't it be better with `""`?

File: helmfile.d/values/openstack-monitoring.yaml.gotmpl [hunk 4]
```diff
@@ -1,2 +1,12 @@
...
+serviceMonitor:
+  {{- if .Values | get "clusterApi.enabled" false }}
+  labels:
+    app: openstack-cloud-controller-manager
+  namespace: openstack-system
+  {{- else }}
+  labels:
+    k8s-app: openstack-cloud-controller-manager
+  namespace: kube-system
+  {{- end }}
```
Comment (Xartos) [inline helmfile.d/values/openstack-monitoring.yaml.gotmpl:12]: Suggestion: Can be simplified with:

```suggestion
serviceMonitor:
  labels:
    {{- if .Values | get "clusterApi.enabled" false }}
    app: openstack-cloud-controller-manager
    {{- else }}
    k8s-app: openstack-cloud-controller-manager
    {{- end }}
  namespace: {{ .Release.Namespace }}
```

File: helmfile.d/charts/openstack-monitoring/values.yaml (hunk 3 above)
Comment (crssnd) [reply helmfile.d/charts/openstack-monitoring/values.yaml]: yes, I fixed it

File: helmfile.d/values/openstack-monitoring.yaml.gotmpl (hunk 4 above)
Comment (crssnd) [reply helmfile.d/values/openstack-monitoring.yaml.gotmpl]: fixed

**Review by davidumea — APPROVED** [review]

**Review by Xartos — APPROVED** [review]

**Pull request merged**